frame/nfts/src/common_functions.rs

```diff
@@ -30,6 +31,27 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Collection::<T, I>::get(collection).map(|i| i.owner)
 	}
 
+	/// Validate the `data` was signed by `signer` and the `signature` is correct.
+	pub fn validate_signature(
+		data: &Vec<u8>,
+		signature: &T::OffchainSignature,
+		signer: &T::AccountId,
+	) -> DispatchResult {
+		// NOTE: for security reasons modern UIs implicitly wrap the data requested to sign into
+		// <Bytes></Bytes>, that's why we support both wrapped and raw versions.
+		let mut wrapped: Vec<u8> = Vec::new();
```

Comment:
You could pre-allocate here to avoid allocations when extending.

Reply:
Done